projects/components/src/search-box/search-box.component.ts

```diff
@@ -69,11 +69,11 @@ export class SearchBoxComponent implements OnInit, OnChanges {
   }
 
   public onSubmit(): void {
-    this.submit.emit(this.value);
+    this.submit.emit(this.value.trim());
   }
 
   public onValueChange(): void {
-    this.debouncedValueSubject.next(this.value);
+    this.debouncedValueSubject.next(this.value.trim());
```

Comment:
we should be doing `this.value = this.value.trim()` here and then emit so that the change reflects in the input box as well.

Reply:
We do not want that.

Reply:
If we do that it will not even let us put a space, right?

Reply:
it shouldn't prevent you from entering spaces unless we are triggering CD cycle right after.